### PR TITLE
Add Beaverctl client and server. #320

### PR DESCRIPTION
**Pull request by Babbleshack**
Adds beavercontrol framework and some simple commands for querying current running status of beaver log shipper, currently allows the client to query the size of log events queue and check whether the transport mechanism is connected.
- run_queue function is now a class.
- I am having trouble testing setup tool installation on my box, if someone could test that it would be great!
- This also hits feature: #320

### CODE REVIEW

**Comment by josegonzalez** [comment]
The whole thing is a merge conflict, so you're going to need to sort those out...

**Comment by Babbleshack** [comment]
I imagine thats because I remove run_queue function, and replace it with RunQueue class

**Comment by Babbleshack** [comment]
merged with master.

**Comment by Babbleshack** [comment]
this commit fails on a Kafka_transport test, which I haven't changed,

```
Traceback (most recent call last):
  File "/home/travis/build/josegonzalez/python-beaver/beaver/tests/test_kafka_transport.py", line 72, in test_builtin_kafka
    cls.assertEqual(n, messages.__len__())
AssertionError: 100 != 50
```

**Comment by Babbleshack** [comment]
```
....Exception in thread Thread-5:
Traceback (most recent call last):
  File "/opt/python/2.7.9/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/opt/python/2.7.9/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/travis/build/josegonzalez/python-beaver/beaver/transports/rabbitmq_transport.py", line 121, in _connection_start
    self._connection.ioloop.start()
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/pika/adapters/select_connection.py", line 135, in start
    self.poller.open = True
AttributeError: 'NoneType' object has no attribute 'open'
.....

Again this is not my code, I am not sure why these tests are failing, perhaps the setup script is broken?

**Comment by Babbleshack** [comment]
conflicts resolved and the branch now merges

**Comment by Babbleshack** [comment]
added documentation for beaverctl

File: docs/user/usage.rst [hunk 1]
```diff
@@ -38,6 +38,24 @@ optional arguments::
     -v, --version         output version and quit
     --fqdn                use the machine's FQDN for source_host
 
+Beaverctl Usage
+---------------
+
+Beaverctl allows users of beaver to monitor the running status of a beaver
+instance from the same machine. Users are avke to query the
```
Comment (JamieCressey) [inline docs/user/usage.rst]: Typo - "Users are able..."

**Comment by JamieCressey** [comment]
Looks good, can you fix the conflicts and add some tests please and we'll get this merged!

**Comment by Babbleshack** [comment]
No problem, Ill need to get a testing env up and running, I haven't looked at this in a long time.

**Comment by JamieCressey** [comment]
Ping @Babbleshack any chance of adding those tests?

**Comment by Babbleshack** [comment]
@JamieCressey Will do this soon, I am finishing some work for uni at the moment.

**Comment by JamieCressey** [comment]
Closing due to inactivity. Feel free to reopen!

**Pull request closed without merging**